packages/cli/src/runners/task-managers/data-request-response-builder.ts

```diff
@@ -0,0 +1,199 @@
+			executionTimeoutTimestamp: additionalData.executionTimeoutTimestamp,
+			restartExecutionId: additionalData.restartExecutionId,
+			userId: additionalData.userId,
+		};
+	}
+
+	buildExecuteData(executeData: IExecuteData | undefined): IExecuteData | undefined {
+		if (executeData === undefined) {
+			return undefined;
+		}
+
+		return {
+			node: executeData.node, // The current node being executed
+			data: this.requestParams.input ? executeData.data : {},
+			source: executeData.source,
+		};
+	}
+
+	buildRunExecutionData(runExecutionData: IRunExecutionData): IRunExecutionData {
+		if (this.requestParams.dataOfNodes === 'all') {
+			return runExecutionData;
+		}
+
+		return {
+			startData: runExecutionData.startData,
+			resultData: {
+				error: runExecutionData.resultData.error,
+				lastNodeExecuted: runExecutionData.resultData.lastNodeExecuted,
+				metadata: runExecutionData.resultData.metadata,
+				runData: this.buildRunData(runExecutionData.resultData.runData),
+				pinData: this.buildPinData(runExecutionData.resultData.pinData),
+			},
+			executionData: runExecutionData.executionData
+				? {
+						// TODO: Figure out what these two are and can they be filtered
+						contextData: runExecutionData.executionData?.contextData,
+						nodeExecutionStack: runExecutionData.executionData.nodeExecutionStack,
+
+						metadata: runExecutionData.executionData.metadata,
+						waitingExecution: runExecutionData.executionData.waitingExecution,
+						waitingExecutionSource: runExecutionData.executionData.waitingExecutionSource,
+					}
+				: undefined,
+		};
+	}
+
+	buildRunData(runData: IRunData): IRunData {
+		return this.filterObjectByNodeNames(runData);
+	}
+
+	buildPinData(pinData: IPinData | undefined): IPinData | undefined {
+		return pinData ? this.filterObjectByNodeNames(pinData) : undefined;
+	}
+
+	buildEnvProviderState(envProviderState: EnvProviderState): EnvProviderState {
+		if (this.requestParams.env) {
+			return envProviderState;
+		}
+
+		return {
+			env: {},
+			isEnvAccessBlocked: envProviderState.isEnvAccessBlocked,
+			isProcessAvailable: envProviderState.isProcessAvailable,
+		};
+	}
+
+	buildInputData(inputData: ITaskDataConnections): ITaskDataConnections {
+		if (this.requestParams.input) {
+			return inputData;
+		}
+
+		return {};
+	}
+
+	buildConnectionInputData(connectionInputData: INodeExecutionData[]): INodeExecutionData[] {
+		if (this.requestParams.input) {
+			return connectionInputData;
+		}
+
+		return [];
+	}
+
+	private buildWorkflow(workflow: Workflow): Omit<WorkflowParameters, 'nodeTypes'> {
+		return {
+			id: workflow.id,
+			name: workflow.name,
+			active: workflow.active,
+			connections: workflow.connectionsBySourceNode,
+			nodes: Object.values(workflow.nodes),
+			pinData: workflow.pinData,
+			settings: workflow.settings,
+			staticData: workflow.staticData,
+		};
+	}
+
+	/**
+	 * Assuming the given `obj` is an object where the keys are node names,
```

Comment:
Would something break if the string is not a valid node name? e.g. user typo

Reply:
Good question. In that case there is no node with that name and filtering works as expected